src/sentry/rules/actions/utils.py

```diff
@@ -0,0 +1,114 @@
+from collections import defaultdict
+from typing import Any, DefaultDict
+
+from sentry.api.serializers.models.rule import generate_rule_label
+from sentry.models.environment import Environment
+from sentry.models.rule import Rule
+
+
+def get_updated_rule_data(rule: Rule) -> dict[str, Any]:
+    rule_data = dict(rule.data)
+    if rule.environment_id:
+        rule_data["environment_id"] = rule.environment_id
+    if rule.owner:
+        rule_data["owner"] = rule.owner
+    rule_data["label"] = rule.label
+    return rule_data
+
+
+def check_value_changed(
+    present_state: dict[str, Any], prior_state: dict[str, Any], key: str, word: str
+) -> str | None:
+    if present_state.get(key) != prior_state.get(key):
+        old_value = prior_state.get(key)
+        new_value = present_state.get(key)
+        return f"Changed {word} from *{old_value}* to *{new_value}*"
+    return None
```

Comment:
small nit: can skip returning None

Reply:
mypy wouldn't let me 😭

Reply:
apparently by design: https://github.com/python/mypy/issues/7511#issuecomment-689667508

There's a flag `--no-warn-no-return` to disable this but looks like we don't have it on